### DISCUSSION TITLE
ColorChooser. How can I change the values?

### DISCUSSION THREAD

**Discussion by ConsoleC137** [fltk-rs/fltk-rs#1303] [Q&A] [answered]
Hello, I'm using `fltk::group::ColorChooser` in my project and I would like to be able to change its values (r, g b), how can I do that? I want this widget to be created not in black (as it is now), but with the colors that I have set
![изображение_2022-10-29_233838036](https://user-images.githubusercontent.com/66591395/198843007-7c7481b6-c2d5-4574-89b0-2f7e16baee6e.png)

P.S. I know about the existence of `fltk_sys::dialog::Fl_color_chooser`, but this function creates a new window with `ColorChooser` and it doesn't allow me to use the callback

**Comment by MoAlyousef** [comment, marked as answer]
This isn’t currently exposed in the wrapper. I’ll add it to the master branch later this evening and should be available in the next release.

**Reply by ConsoleC137** [reply]
Thanks! It will be perfect for me. I tried to do it on my own and even started studying the source code, but I realized that I wouldn't do it on my own.

**Reply by MoAlyousef** [reply]
This has been added to master:
```rust
    let mut cc = group::ColorChooser::new(0, 0, 400, 300, None);
    cc.set_rgb(255, 255, 255).ok();
```
The next release will hopefully be on the 1st of November.

**Reply by ConsoleC137** [reply]
I have a problem. I can't use the master branch. Below I left a screenshot of the error. How can I fix this?

![изображение_2022-10-30_174931007](https://user-images.githubusercontent.com/66591395/198874671-e622a84a-13ea-491b-adf7-6dc2620d915e.png)

**Reply by MoAlyousef** [reply]
The master branch won’t work currently with fltk-bundled since there is no bundle for version 1.3.19 (unreleased yet). 
It would require cmake and a c++ compiler. 
Otherwise, you can manually grab a lib_x64-windows-msvc.zip bundle from here:

https://github.com/MoAlyousef/fltk_prebuilds/actions/runs/3355417785

unzip and navigate inside, copy the lib folder to your project. 
Then build against it using:
```
CFLTK_BUNDLE_DIR=C:\Rust\shutdown_manager cargo build
```

**Reply by ConsoleC137** [reply]
Sorry to bother you. I have done all of the above, but I do not know where to insert this line `CFLTK_BUNDLE_DIR=C:\Rust\shutdown_manager cargo build`. 
![image](https://user-images.githubusercontent.com/66591395/198884126-4d8674fd-e8fa-40b1-9328-bd6f82f4fe89.png)
Here is the current status of my project. I will be grateful to you for any help you have given me.

**Reply by MoAlyousef** [reply]
Are you building using the command-line?

**Reply by MoAlyousef** [reply]
I guess with powershell it would be:
`$env:CFLTK_BUNDLE_DIR="C:\Rust\shutdown_manager"; cargo build`

**Reply by ConsoleC137** [reply]
Thank you for answering me. I executed the command you wrote above, but I still got an error, what is my mistake?
![изображение_2022-10-30_221636295](https://user-images.githubusercontent.com/66591395/198886434-3128d69d-0ccd-42d9-96f0-fb0c1f634be0.png)

**Reply by MoAlyousef** [reply]
what's the content of the lib folder?
It should directly have the *.lib files.

**Reply by ConsoleC137** [reply]
There are only .lib files in this folder
![изображение_2022-10-30_222512997](https://user-images.githubusercontent.com/66591395/198886875-eae033b1-9db6-44ba-86e3-326d8aae6006.png)

**Reply by MoAlyousef** [reply]
That’s strange. It seems it’s not finding the path and still tries to download the bundle. You can try for example replacing with "." or escaping the backslashes `C:\\Rust\\shutdown_manager`.

**Reply by ConsoleC137** [reply]
I tried different options, but all my attempts were in vain. However, I decided to copy the files directly into my project and it helped me! For some reason cargo ignores this variable. Thank you so much for your help.
![изображение_2022-10-30_230828830](https://user-images.githubusercontent.com/66591395/198889047-737d21fd-c08f-4bb2-b85e-95fb12f250af.png)

![изображение_2022-10-30_230714511](https://user-images.githubusercontent.com/66591395/198888987-02bf1c7c-458a-4f59-b495-d1f35eff667e.png)
![изображение_2022-10-30_230739589](https://user-images.githubusercontent.com/66591395/198889011-2c231fa2-4aa1-42f4-9273-82eebbf6e1cf.png)
